Media/RTShaderLib/SGXLib_NormalMap.glsl

```diff
@@ -58,11 +84,79 @@ void SGX_Generate_Parallax_Texcoord(in sampler2D normalHeightMap,
 						in vec2 texCoord,
 						in vec3 viewPos,
 						in vec2 scaleBias,
+						in mat3 TBN,
+						out vec2 newTexCoord)
+{
+	//Calculate eye direction
+	vec3 eyeVec = mul(-viewPos, TBN);
+	eyeVec = normalize(eyeVec);
+
+	//Simple parallax mapping
+	newTexCoord = texCoord;
+	float height = 1.0f - texture2D(normalHeightMap, newTexCoord).a;
+
+	#ifndef TERRAIN_PARALLAX_MAPPING
+		eyeVec.y = -eyeVec.y; //Inverse y
+		vec2 p = eyeVec.xy / eyeVec.z * (height * scaleBias.x);
+	#else
+	    vec2 p = eyeVec.xy * (height * scaleBias.x);
+	#endif
+
+    newTexCoord = newTexCoord - p;
+}
+
+void SGX_Generate_Parallax_Steep_Texcoord(in sampler2D normalHeightMap,
+						in vec2 texCoord,
+						in vec3 viewPos,
+						in vec2 scaleBias,
+						in float layerCount,
+						in float maxDistance,
```

Comment:
this parameter is no longer used now

Reply:
Ooops! Fixed.

Reply:
I did not even notice `scaleBias`.. I was referring to `maxDistance`

Reply:
> I did not even notice `scaleBias`.. I was referring to `maxDistance`

Ah! Sorry, it should be used - the conditional was removed somehow. Framerate plummets for no good reason with an infinate distance as you cant see it from a distance anyway. Ill add this back in.